src/updaters/java-auth-readme.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { checkpoint, CheckpointType } from '../util/checkpoint';
+import { Update, UpdateOptions } from './update';
+import { GitHubFileContents } from '../github';
+
+export class JavaAuthReadme implements Update {
+  path: string;
+  changelogEntry: string;
+  version: string;
+  versions?: { [key: string]: string };
+  packageName: string;
+  create: boolean;
+  contents?: GitHubFileContents;
+
+  constructor(options: UpdateOptions) {
+    this.create = false;
+    this.path = options.path;
+    this.changelogEntry = options.changelogEntry;
+    this.version = options.version;
+    this.packageName = options.packageName;
+  }
+  updateContent(content: string): string {
+    content = content.replace(
+      /<version>.*<\/version>/,
+      `<version>${this.version}</version>`
+    );
+    content = content.replace(
+      /'com.google.auth:google-auth-library-oauth2-http:[^']+'/,
```

Comment:
If you're looking to generalize this behavior, we do tag lines to search for versions:
* `{x-version-update:google-auth-library-bom:current}`: replace any Java semver pattern on that same line with the `current-version` of `google-auth-library-bom` found in the `versions.txt` file.
* Between `{x-version-update-start:google-auth-library-oauth2-http:released}` and `{x-version-update-end}`: replace any Java semver pattern with the `released-version` of `google-auth-library-oauth2-http` found in the `versions.txt` file.

In theory, with these two rules, you could run it on any file, rather than having per-file rules. In gradle setups, there are different build files that specify versions.

Reply:
👍 might hold off and make the logic generic as we do the repo-split. I think we'll potentially  want to pass `groupId` into the updated? so that it can use this along with `packageName` to create a more generic replacement?

Are we going to have similar lines in the README for each split repo?

Reply:
Yes, the plan is to keep the README for each google-cloud-[service] library as the main README on the split repo.